WombatProject/Services/GoogleBooksAPIService.cs

```diff
@@ -7,58 +8,64 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 using WombatLibrarianApi.Models;
+using WombatLibrarianApi.Settings;
 
 namespace WombatLibrarianApi.Services
 {
-    public class GoogleBooksAPIService : BookAPIService
+    public class GoogleBooksAPIService : IBookAPIService
     {
-        private readonly IConfiguration _configuration;
-        public WombatBooksContext Context { get; }
-        public List<Book> AuthorBookItems { get; set; } = new List<Book>();
-        public List<Book> BookItems { get; set; } = new List<Book>();
+        protected readonly GoogleApiSettings _googleApiSettings;
+        public List<Book> AuthorBookItems { get; } = new List<Book>();
+        public List<Book> SearchResults { get; } = new List<Book>();
 
-        public GoogleBooksAPIService(IConfiguration configuration, WombatBooksContext context)
+        public GoogleBooksAPIService(IOptions<GoogleApiSettings> settings)
         {
-            _configuration = configuration;
-            Context = context;
+            this._googleApiSettings = settings.Value;
         }
 
-        public async Task GetSearchResults(string searchTerm)
+        public async Task GetSearchResultsAsync(string searchTerm)
         {
-            BookItems.Clear();
-            string url = $"{_configuration["GBooksURL"]}?q={searchTerm}&maxResults=40";
+            SearchResults.Clear();
+            string url = $"{_googleApiSettings.GoogleBooksURL}?q={searchTerm}&maxResults=40";
             IList<JToken> tokens = await GetBookItemsAsJToken(url);
             foreach (JToken token in tokens)
             {
-                BookItems.Add(parseJsonToken(token));
+                SearchResults.Add(ParseJsonToken(token));
             }
         }
 
-        public async Task GetAuthorBooks(string author)
+        public async Task GetAuthorBooksAsync(string author)
         {
             AuthorBookItems.Clear();
-            string url = $"{_configuration["GBooksURL"]}?q=inauthor:{author}";
+            string url = $"{_googleApiSettings.GoogleBooksURL}?q=inauthor:{author}";
             IList<JToken> tokens = await GetBookItemsAsJToken(url);
             foreach (JToken token in tokens)
             {
-                AuthorBookItems.Add(parseJsonToken(token));
+                AuthorBookItems.Add(ParseJsonToken(token));
             }
         }
 
         private async Task<IList<JToken>> GetBookItemsAsJToken(string url)
         {
-            using (var client = new HttpClient())
+            var client = new HttpClient();
+            var uri = new Uri(url);
+            var response = await client.GetAsync(uri);
+            string textResult = await response.Content.ReadAsStringAsync();
+            JObject bookSearch = JObject.Parse(textResult);
```

Comment:
Use `JSONSerializer` instead of JObject. You can see example in the documentation page linked above.

Reply:
May I ask why?

Reply:
I did as you suggested but I have to add that I had to refactor half of the project to use JSONSerializer instead of JObject and it took a LOT of work ... but it's done, hopefully will make unit testing easier